discovery/manager.go

```diff
@@ -115,6 +115,13 @@ func (m *Manager) ApplyConfig(cfg map[string]sd_config.ServiceDiscoveryConfig) e
 	return nil
 }
 
+// StartUnknownProvider is used for sdtool. Only use this if you know what you're doing.
+func (m *Manager) StartUnknownProvider(ctx context.Context, name string, worker Discoverer) {
+	// Pool key for non-standard SD implementations are unknown.
+	poolKey := poolKey{setName: name, provider: name}
+	m.startProvider(ctx, poolKey, worker)
+}
+
```

Comment:
is there no way to do this without modifying the Discovery manager?

Reply:
Not that I can see. SD providers are normally started via `providersFromConfig` which only knows about existing SD types. Also, I think it's best if knowledge of `poolKey` is contained to `Manager`.

Reply:
I will try to refactor the SD a bit to allow its usage without such modification.
I am thinking to implement the Discoverers as plugable middleware so that this can be done at at the initialisation.